.travis.yml

```diff
@@ -13,7 +13,7 @@ cache:
     - $HOME/.cache/pip/wheels
 
 addons:
-  postgresql: "9.4"
+  postgresql: "9.5"
```

Comment:
what's the logic here, ooi? Given we have to pick one version of postgres to test against, it feels like we should be testing against the oldest one we support.

Reply:
oh I see, it's because postgres 9.4 doesn't support upsert. hrm, we need to figure out what to do here

Reply:
As I just wrote in #synapse-dev:

> I am sorely tempted to say that we should do both [9.4 and 9.5], and the solution to the pain that it will cause is to either do our CI better or drop support for 9.4

Reply:
The reason why I want 9.5 here is that we need to test the native upsert path, which can use the postgres-specific batch code.

We should do our CI better and support more postgreses, but if we have to pick one, 9.5 is the one that covers the most code.